src/Pos/Wallet/Web/Server/Sockets.hs

```diff
@@ -91,5 +109,52 @@ type WebWalletSockets m = (MonadWalletWebSockets m, MonadIO m)
 runWalletWS :: ConnectionsVar -> WalletWebSockets m a -> m a
 runWalletWS = flip Ether.runReaderT
 
-notify :: WebWalletSockets m => NotifyEvent -> m ()
-notify msg = getWalletWebSockets >>= flip sendWS msg
+notifyAll :: WebWalletSockets m => NotifyEvent -> m ()
+notifyAll msg = do
+  var <- getWalletWebSockets
+  conns <- readTVarIO var
+  for_ (listConnections conns) $ flip sendWS msg
+
+------------------------------------
+-- Implementation of ConnectionSet
+------------------------------------
+
+data TaggedSet v = TaggedSet
+    { tsUnusedTag :: IM.Key
+    , tsData      :: IntMap v
+    }
+
+instance Default (TaggedSet v) where
+  def = TaggedSet 0 IM.empty
+
+type ConnectionTag = IM.Key
+type ConnectionSet = TaggedSet WSConnection
+
+registerConnection :: WSConnection -> ConnectionsVar -> STM ConnectionTag
+registerConnection conn var = modifyTVarS var $ do
+    conns <- get
```

Comment:
It's actually `modifyTVarS var $ StateT $ pure . registerConnectionPure conn`, but honestly I don't think this is more readable. Do you?